google/field_helpers.go

```diff
@@ -77,6 +82,86 @@ func parseGlobalFieldValue(resourceType, fieldValue, projectSchemaField string,
 	}, nil
 }
 
+type ZonalFieldValue struct {
+	Project string
+	Zone    string
+	Name    string
+
+	resourceType string
+}
+
+func (f ZonalFieldValue) RelativeLink() string {
+	if len(f.Name) == 0 {
+		return ""
+	}
+
+	return fmt.Sprintf(zonalLinkTemplate, f.Project, f.Zone, f.resourceType, f.Name)
+}
+
+// Parses a zonal field supporting 5 different formats:
+// - https://www.googleapis.com/compute/ANY_VERSION/projects/{my_project}/zones/{zone}/{resource_type}/{resource_name}
+// - projects/{my_project}/zones/{zone}/{resource_type}/{resource_name}
+// - zones/{zone}/{resource_type}/{resource_name}
+// - resource_name
+// - "" (empty string). RelativeLink() returns empty if isEmptyValid is true.
+//
+// If the project is not specified, it first tries to get the project from the `projectSchemaField` and then fallback on the default project.
+// If the zone is not specified, it takes the value of `zoneSchemaField`.
+
+func parseZonalFieldValue(resourceType, fieldValue, projectSchemaField, zoneSchemaField string, d TerraformResourceData, config *Config, isEmptyValid bool) (*ZonalFieldValue, error) {
+	if len(fieldValue) == 0 {
+		if isEmptyValid {
+			return &ZonalFieldValue{resourceType: resourceType}, nil
+		}
+		return nil, fmt.Errorf("The zonal field for resource %s cannot be empty.", resourceType)
+	}
+
+	r := regexp.MustCompile(fmt.Sprintf(zonalLinkBasePattern, resourceType))
+	if r.MatchString(fieldValue) {
+		parts := r.FindStringSubmatch(fieldValue)
+
+		return &ZonalFieldValue{
+			Project:      parts[1],
+			Zone:         parts[2],
+			Name:         parts[3],
+			resourceType: resourceType,
+		}, nil
+	}
+
+	project, err := getProjectFromSchema(projectSchemaField, d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	r = regexp.MustCompile(fmt.Sprintf(zonalPartialLinkBasePattern, resourceType))
+	if r.MatchString(fieldValue) {
```

Comment:
You might be able to collapse this and the following line via:
```go
if parts := r.FindStringSubmatch(fieldValue); parts != nil {
   // stuff happens
}
```

Reply:
Done